airflow/models.py

```diff
@@ -764,6 +770,8 @@ def command(
         cmd += "--local " if local else ""
         cmd += "--pool {pool} " if pool else ""
         cmd += "--raw " if raw else ""
+        cmd += "--airflow_cfg \"{pickled_cfg}\" " if airflow_cfg else ""
```

Comment:
I think the airflow.cfg should be the same for all tasks (i.e. tasks should not rely on any entry in airflow.cfg). Any custom secrets (e.g. fernet key, the current "connections" table, etc) that tasks use should be stored separately, and only accessible by the right users/groups. This requires pulling some of the config out of airflow.cfg and building a secret storing system similar to e.g. http://kubernetes.io/docs/user-guide/secrets/ .